packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -18,126 +18,127 @@ import H from '../constants';
 const watchmanURL =
   'https://facebook.github.io/watchman/docs/troubleshooting.html';
 
-function isDescendant(root: string, child: string): boolean {
-  return child.startsWith(root);
+class WatchmanError extends Error {
```

Comment:
Reason for this overhaul was to make stack traces useful since otherwise I always ended up in the helper function which was far from being helpful.

Reply:
Not a fan of overriding the `stack` field. Can you just mutate the message of the original error instance instead?